### PR TITLE
Support location widget to get current location

### LINKED ISSUE #800: Add location mapping widget
**Is your feature request related to a problem? Please describe.**
Add a location widget that allows you to submit the location with GPS coordinates and location details as FHIR [Location](https://www.hl7.org/fhir/location.html) resources
**Describe the solution you'd like**
1. **ANC Usecase**
 In the ANC use case, add a location of a facility or a family residence or village during a family registration/enrollment [Encounter](https://www.hl7.org/fhir/encounter.html#Encounter) that will in turn reference a Patient plus a Practioner. Tagged via [Encounter.location](https://www.hl7.org/fhir/encounter-definitions.html#Encounter.location.location) This should submit a location payload tagged to a patient while enrolling ANC Clients or registering a family and its householder members
2. **Covax Usecase**
In the Covax demo app, allow a user to submit the location details of where the Vaccination is being done

The Widget should be able to trigger the GPS to collect Latitude and Longitude coordinates for a precise location 

_Sample location payload for an Immunization facility_
```
{
  "resourceType": "Location",
  "status": "active",
  "name": "Arundel mobile clinic",
  "description": "Vaccination Site",
  "mode": "instance",
  "address": {
    "city": "Nairobi",
    "state": "MD",
    "country": "USA"
  },
  "physicalType": {
    "coding": [ {
      "system": "http://terminology.hl7.org/CodeSystem/location-physical-type",
      "code": "bu",
      "display": "Vaccination Center"
    } ]
  },
  "position": {
    "longitude": 42.256500,
    "latitude": -83.694710
  }
}
```

**Describe alternatives you've considered**
N/A

**Additional context**
_**Sample location payload for an Immunization facility**_ 

```
{
  "resourceType": "Location",
  "status": "active",
  "name": "Arundel mobile clinic",
  "description": "Vaccination Site",
  "mode": "instance",
  "address": {
    "city": "Nairobi",
    "state": "MD",
    "country": "USA"
  },
  "physicalType": {
    "coding": [ {
      "system": "http://terminology.hl7.org/CodeSystem/location-physical-type",
      "code": "bu",
      "display": "Vaccination Center"
    } ]
  },
  "position": {
    "longitude": 42.256500,
    "latitude": -83.694710
  }
}
```

_**Sample location payload for a Household location/home**_ 

```
{
  "resourceType": "Location",
  "status": "active",
  "name": "Kasera Village",
  "description": "The home village location",
  "mode": "instance",
  "address": {
    "city": "Nairobi",
    "state": "Kenya",
    "country": "Kenya"
  },
  "physicalType": {
    "coding": [ {
      "system": "http://terminology.hl7.org/CodeSystem/location-physical-type",
      "code": "bu",
      "display": "Kasera Village"
    } ]
  },
  "position": {
    "longitude": 42.256500,
    "latitude": -83.694710
  }
}
```

**_Sample ANC mock_** 
Screen Shot 2021-09-07 at 11 14 23 AM

**_NB. The accompanying details of the Location (address, display name) can be collected via a questionnaire and appended to the Questionnaire response before extraction and saving/posting/sync as a location resource_**

### PR DESCRIPTION
**Pull request by LZRS**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #800 

**Description**
Add a custom component, location-widget in contrib/locationwidget, to detect and pick current location coordinates to populate latitude/longitude questionnaire fields

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Feature

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by LZRS** [comment]
Questionnaire
``` json

{
  "resourceType": "Questionnaire",
  "language": "en",
  "status": "active",
  "date": "2020-11-18T07:24:47.111Z",
  "item": [
    {
      "linkId": "location-widget",
      "type": "group",
      "text": "Location Widget",
      "extension": [
        {
          "url": "https://github.com/google/android-fhir/StructureDefinition/questionnaire-itemControl",
          "valueCodeableConcept": {
            "coding": [
              {
                "system": "https://github.com/google/android-fhir/questionnaire-item-control",
                "code": "location-widget"
              }
            ]
          }
        }
      ],
      "item": [
        {
          "linkId": "latitude",
          "type": "decimal",
          "required": true,
          "text": "Latitude",
          "extension": [
            {
              "url": "gps-coordinate",
              "valueString": "latitude"
            }
          ]
        },
        {
          "linkId": "longitude",
          "type": "decimal",
          "required": true,
          "text": "Longitude",
          "extension": [
            {
              "url": "gps-coordinate",
              "valueString": "longitude"
            }
          ]
        }
      ]
    }
  ]
}


```

**Review by jingtang10 — APPROVED** [review]

**Comment by f-odhiambo** [comment]
Can we merge this in ? @jingtang10

**Review by jingtang10 — APPROVED** [review]

File: contrib/locationwidget/src/test/java/com/google/android/fhir/datacapture/contrib/views/locationwidget/ExampleUnitTest.kt
Comment (jingtang10) [inline contrib/locationwidget/src/test/java/com/google/android/fhir/datacapture/contrib/views/locationwidget/ExampleUnitTest.kt:1]: remove this file

File: contrib/locationwidget/src/androidTest/java/com/google/android/fhir/datacapture/contrib/views/locationwidget/ExampleInstrumentedTest.kt
Comment (jingtang10) [inline contrib/locationwidget/src/androidTest/java/com/google/android/fhir/datacapture/contrib/views/locationwidget/ExampleInstrumentedTest.kt:1]: is this file actually testing anything? if not we can remove.

File: datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireItemComponents.kt [hunk 1]
```diff
@@ -183,31 +183,32 @@ val Questionnaire.QuestionnaireItemComponent.initialExpression: Expression?
...
   }
 
-/**
- * The [ItemControlTypes] of the questionnaire item if it is specified by the item control
- * extension, or `null`.
- *
- * See http://hl7.org/fhir/R4/extension-questionnaire-itemcontrol.html.
- */
-val Questionnaire.QuestionnaireItemComponent.itemControl: ItemControlTypes?
+val Questionnaire.QuestionnaireItemComponent.uiControlControlCode: String?
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireItemComponents.kt:186]: i actually don't love the name of this property. can we call it `itemControlCode`?

also a little bit strange to use the item control extension together with any custom extension we have defined in android fhir sdk...  but i suppose that's a problem with the old code already. so not necessarily have to be addressed in this pr.

**Review by jingtang10 — APPROVED** [review]
please feel free to merge once the comments below are addressed.

thanks @LZRS

**Comment by jingtang10** [comment]
please go through the check list in the pr description @LZRS

**Pull request merged**